module/Motion/DistanceStraight.cpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * @file   DistanceStraight.cpp
+ * @brief  目標距離まで直進するクラス
+ * @author bizyutyu
+ */
+
+#include "DistanceStraight.h"
+using namespace std;
+
+DistanceStraight::DistanceStraight(double _targetDistance, double _targetSpeed)
+  : Straight(_targetSpeed), targetDistance(_targetDistance)
+{
+}
+
+bool DistanceStraight::isMetPreCondition()
+{
+  char buf[LARGE_BUF_SIZE];
+  // 目標速度値が0の場合は終了する
+  if(targetSpeed == 0) {
+    logger.logWarning("The targetSpeed value passed to DistanceStraight is 0");
+    return false;
+  }
+  // 目標距離の値が0以下の場合はwarningを出して終了する
+  if(targetDistance <= 0) {
+    snprintf(buf, LARGE_BUF_SIZE, "The targetDistance value passed to DistanceStraight is %.2f",
+             targetDistance);
+    logger.logWarning(buf);
+    return false;
+  }
+
+  return true;
+}
+
+bool DistanceStraight::isMetContinuationCondition()
+{
+  // 現在の走行距離を取得する
+  currentRightMotorCount = measurer.getRightCount();
+  currentLeftMotorCount = measurer.getLeftCount();
+  currentRightDistance = Mileage::calculateWheelMileage(currentRightMotorCount);
+  currentLeftDistance = Mileage::calculateWheelMileage(currentLeftMotorCount);
+
+  // 現在の走行距離が目標走行距離に達していなければtrueを返す
+  // 左右どちらとも
+  if((abs(currentRightDistance - initialRightDistance) < targetDistance)
+     && (abs(currentLeftDistance - initialLeftDistance) < targetDistance)) {
+    return true;
+  }
+
+  // 現在の走行距離が目標走行距離に達した場合falseを返す
+  return false;
```

Comment:
ここも同じです．あと，変数*Distanceがdouble型になっているので，fabsの方がよいかもしれないです．

```suggestion
  // 現在の走行距離が目標走行距離に達していればfalseを返す
  // 左右どちらか
  if((fabs(Mileage::calculateWheelMileage(currentRightMotorCount) - initialRightDistance) >= targetDistance)
     || (abs(Mileage::calculateWheelMileage(currentLeftMotorCount) - initialLeftDistance) >= targetDistance)) {
    return false;
  }

  // 現在の走行距離が目標走行距離に達していない場合trueを返す
  return true;
```

ついでに確認したいのですが，DistanceLineTracingの継続条件では，
```
if(fabs(Mileage::calculateMileage(measurer.getRightCount(), measurer.getLeftCount())
- initDistance)
return false;
```
のように，車輪ごとに判定するのではなく平均して判定するようにしているのですが，どちらがよいですか？
合わせておきたいです．